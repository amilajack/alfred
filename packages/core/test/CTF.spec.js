/* eslint no-restricted-syntax: off */
import path from 'path';
import powerset from '@amilajack/powerset';
import CTF, {
  CTFS,
  getConfigs,
  getDependencies,
  writeConfigsFromCtf,
  getDepsInstallCommand
} from '@alfredpkg/core';

describe('CTF', () => {
  describe('executors', () => {
    let ctf;

    beforeAll(() => {
      ctf = new Map();
      ctf.set('webpack', CTFS.webpack);
    });

    it('should run simple executor', async () => {
      await writeConfigsFromCtf(ctf);
      const fixturePath = path.join(__dirname, 'fixtures', 'test-alfred-app');
      const execDepsScript = getDepsInstallCommand(
        ['alfred-skill-webpack'],
        fixturePath
      );
      expect(execDepsScript).toMatchSnapshot();
    });
  });

  // Generate tests for CTF combinations
  const ctfNamesCombinations = powerset(Object.keys(CTFS)).sort();
  for (const ctfCombination of ctfNamesCombinations) {
    it(`combination ${ctfCombination.join(',')}`, () => {
      expect(ctfCombination).toMatchSnapshot();
      // Get the CTFs for each combination
      const filteredCtfs = ctfCombination.map(ctfName => CTFS[ctfName]);
      const result = CTF(filteredCtfs);
      expect(getConfigs(result)).toMatchSnapshot();
      expect(getDependencies(result)).toMatchSnapshot();
    });
  }
});
