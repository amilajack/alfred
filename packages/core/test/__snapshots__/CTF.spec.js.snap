// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CTF combination babel 1`] = `
Array [
  "babel",
]
`;

exports[`CTF combination babel 2`] = `
Array [
  Object {
    "extends": Array [
      "@babel/preset-env",
    ],
  },
]
`;

exports[`CTF combination babel 3`] = `Object {}`;

exports[`CTF combination babel 4`] = `
Object {
  "@babel/cli": "7.2.0",
  "@babel/core": "7.2.0",
  "@babel/preset-env": "7.2.0",
}
`;

exports[`CTF combination babel,eslint 1`] = `
Array [
  "babel",
  "eslint",
]
`;

exports[`CTF combination babel,eslint 2`] = `
Array [
  Object {
    "extends": Array [
      "@babel/preset-env",
    ],
  },
  Object {
    "extends": Array [
      "bliss",
    ],
    "parser": "babel-eslint",
  },
]
`;

exports[`CTF combination babel,eslint 3`] = `Object {}`;

exports[`CTF combination babel,eslint 4`] = `
Object {
  "@babel/cli": "7.2.0",
  "@babel/core": "7.2.0",
  "@babel/preset-env": "7.2.0",
  "babel-eslint": "5.0.0",
  "eslint": "5.10.0",
}
`;

exports[`CTF combination babel,eslint,jest 1`] = `
Array [
  "babel",
  "eslint",
  "jest",
]
`;

exports[`CTF combination babel,eslint,jest 2`] = `
Array [
  Object {
    "extends": Array [
      "@babel/preset-env",
    ],
  },
  Object {
    "extends": Array [
      "bliss",
    ],
    "parser": "babel-eslint",
    "plugins": Array [
      "jest",
    ],
  },
  Object {},
]
`;

exports[`CTF combination babel,eslint,jest 3`] = `Object {}`;

exports[`CTF combination babel,eslint,jest 4`] = `
Object {
  "@babel/cli": "7.2.0",
  "@babel/core": "7.2.0",
  "@babel/preset-env": "7.2.0",
  "babel-eslint": "5.0.0",
  "babel-jest": "8.0.0",
  "eslint": "5.10.0",
  "eslint-plugin-jest": "8.0.0",
  "jest": "5.0.0",
}
`;

exports[`CTF combination babel,eslint,prettier 1`] = `
Array [
  "babel",
  "eslint",
  "prettier",
]
`;

exports[`CTF combination babel,eslint,prettier 2`] = `
Array [
  Object {
    "extends": Array [
      "@babel/preset-env",
    ],
  },
  Object {
    "extends": Array [
      "prettier",
    ],
    "parser": "babel-eslint",
    "plugins": Array [
      "prettier",
    ],
  },
  Object {},
]
`;

exports[`CTF combination babel,eslint,prettier 3`] = `
Object {
  "eslint-config-prettier": "3.3.0",
  "eslint-plugin-prettier": "3.0.1",
}
`;

exports[`CTF combination babel,eslint,prettier 4`] = `
Object {
  "@babel/cli": "7.2.0",
  "@babel/core": "7.2.0",
  "@babel/preset-env": "7.2.0",
  "babel-eslint": "5.0.0",
  "eslint": "5.10.0",
  "prettier": "1.15.3",
}
`;

exports[`CTF combination babel,eslint,prettier,jest 1`] = `
Array [
  "babel",
  "eslint",
  "prettier",
  "jest",
]
`;

exports[`CTF combination babel,eslint,prettier,jest 2`] = `
Array [
  Object {
    "extends": Array [
      "@babel/preset-env",
    ],
  },
  Object {
    "extends": Array [
      "prettier",
    ],
    "parser": "babel-eslint",
    "plugins": Array [
      "jest",
    ],
  },
  Object {},
  Object {},
]
`;

exports[`CTF combination babel,eslint,prettier,jest 3`] = `
Object {
  "eslint-config-prettier": "3.3.0",
  "eslint-plugin-prettier": "3.0.1",
}
`;

exports[`CTF combination babel,eslint,prettier,jest 4`] = `
Object {
  "@babel/cli": "7.2.0",
  "@babel/core": "7.2.0",
  "@babel/preset-env": "7.2.0",
  "babel-eslint": "5.0.0",
  "babel-jest": "8.0.0",
  "eslint": "5.10.0",
  "eslint-plugin-jest": "8.0.0",
  "jest": "5.0.0",
  "prettier": "1.15.3",
}
`;

exports[`CTF combination babel,jest 1`] = `
Array [
  "babel",
  "jest",
]
`;

exports[`CTF combination babel,jest 2`] = `
Array [
  Object {
    "extends": Array [
      "@babel/preset-env",
    ],
  },
  Object {},
]
`;

exports[`CTF combination babel,jest 3`] = `Object {}`;

exports[`CTF combination babel,jest 4`] = `
Object {
  "@babel/cli": "7.2.0",
  "@babel/core": "7.2.0",
  "@babel/preset-env": "7.2.0",
  "babel-jest": "8.0.0",
  "jest": "5.0.0",
}
`;

exports[`CTF combination babel,prettier 1`] = `
Array [
  "babel",
  "prettier",
]
`;

exports[`CTF combination babel,prettier 2`] = `
Array [
  Object {
    "extends": Array [
      "@babel/preset-env",
    ],
  },
  Object {},
]
`;

exports[`CTF combination babel,prettier 3`] = `Object {}`;

exports[`CTF combination babel,prettier 4`] = `
Object {
  "@babel/cli": "7.2.0",
  "@babel/core": "7.2.0",
  "@babel/preset-env": "7.2.0",
  "prettier": "1.15.3",
}
`;

exports[`CTF combination babel,prettier,jest 1`] = `
Array [
  "babel",
  "prettier",
  "jest",
]
`;

exports[`CTF combination babel,prettier,jest 2`] = `
Array [
  Object {
    "extends": Array [
      "@babel/preset-env",
    ],
  },
  Object {},
  Object {},
]
`;

exports[`CTF combination babel,prettier,jest 3`] = `Object {}`;

exports[`CTF combination babel,prettier,jest 4`] = `
Object {
  "@babel/cli": "7.2.0",
  "@babel/core": "7.2.0",
  "@babel/preset-env": "7.2.0",
  "babel-jest": "8.0.0",
  "jest": "5.0.0",
  "prettier": "1.15.3",
}
`;

exports[`CTF combination babel,webpack 1`] = `
Array [
  "babel",
  "webpack",
]
`;

exports[`CTF combination babel,webpack 2`] = `
Array [
  Object {
    "extends": Array [
      "@babel/preset-env",
    ],
  },
  Object {
    "entry": "/",
    "mode": "development",
    "module": Object {
      "rules": Array [
        Object {
          "exclude": /node_modules/,
          "test": /\\\\\\.jsx\\?\\$/,
          "use": Object {
            "loader": "babel-loader",
            "options": Object {
              "cacheDirectory": true,
            },
          },
        },
      ],
    },
    "output": Object {
      "filename": "main.js",
      "path": "/",
      "publicPath": "./targets/",
    },
    "plugins": Array [],
    "resolve": Object {
      "extensions": Array [
        ".js",
        ".json",
      ],
    },
  },
]
`;

exports[`CTF combination babel,webpack 3`] = `Object {}`;

exports[`CTF combination babel,webpack 4`] = `
Object {
  "@babel/cli": "7.2.0",
  "@babel/core": "7.2.0",
  "@babel/preset-env": "7.2.0",
  "babel-loader": "5.0.0",
  "webpack": "4.28.3",
  "webpack-cli": "3.2.1",
}
`;

exports[`CTF combination babel,webpack,eslint 1`] = `
Array [
  "babel",
  "webpack",
  "eslint",
]
`;

exports[`CTF combination babel,webpack,eslint 2`] = `
Array [
  Object {
    "extends": Array [
      "@babel/preset-env",
    ],
  },
  Object {
    "entry": "/",
    "mode": "development",
    "module": Object {
      "rules": Array [
        Object {
          "exclude": /node_modules/,
          "test": /\\\\\\.jsx\\?\\$/,
          "use": Object {
            "loader": "babel-loader",
            "options": Object {
              "cacheDirectory": true,
            },
          },
        },
      ],
    },
    "output": Object {
      "filename": "main.js",
      "path": "/",
      "publicPath": "./targets/",
    },
    "plugins": Array [],
    "resolve": Object {
      "extensions": Array [
        ".js",
        ".json",
      ],
    },
  },
  Object {
    "extends": Array [
      "bliss",
    ],
    "parser": "babel-eslint",
    "settings": Object {
      "import/resolver": Object {
        "webpack": Object {
          "config": "configs/webpack.config.js",
        },
      },
    },
  },
]
`;

exports[`CTF combination babel,webpack,eslint 3`] = `Object {}`;

exports[`CTF combination babel,webpack,eslint 4`] = `
Object {
  "@babel/cli": "7.2.0",
  "@babel/core": "7.2.0",
  "@babel/preset-env": "7.2.0",
  "babel-eslint": "5.0.0",
  "babel-loader": "5.0.0",
  "eslint": "5.10.0",
  "webpack": "4.28.3",
  "webpack-cli": "3.2.1",
}
`;

exports[`CTF combination babel,webpack,eslint,jest 1`] = `
Array [
  "babel",
  "webpack",
  "eslint",
  "jest",
]
`;

exports[`CTF combination babel,webpack,eslint,jest 2`] = `
Array [
  Object {
    "extends": Array [
      "@babel/preset-env",
    ],
  },
  Object {
    "entry": "/",
    "mode": "development",
    "module": Object {
      "rules": Array [
        Object {
          "exclude": /node_modules/,
          "test": /\\\\\\.jsx\\?\\$/,
          "use": Object {
            "loader": "babel-loader",
            "options": Object {
              "cacheDirectory": true,
            },
          },
        },
      ],
    },
    "output": Object {
      "filename": "main.js",
      "path": "/",
      "publicPath": "./targets/",
    },
    "plugins": Array [],
    "resolve": Object {
      "extensions": Array [
        ".js",
        ".json",
      ],
    },
  },
  Object {
    "extends": Array [
      "bliss",
    ],
    "parser": "babel-eslint",
    "plugins": Array [
      "jest",
    ],
    "settings": Object {
      "import/resolver": Object {
        "webpack": Object {
          "config": "configs/webpack.config.js",
        },
      },
    },
  },
  Object {
    "moduleNameMapper": Object {
      "\\\\.(css|less|sass|scss)$": "identity-obj-proxy",
      "\\\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/mocks/fileMock.js",
    },
  },
]
`;

exports[`CTF combination babel,webpack,eslint,jest 3`] = `Object {}`;

exports[`CTF combination babel,webpack,eslint,jest 4`] = `
Object {
  "@babel/cli": "7.2.0",
  "@babel/core": "7.2.0",
  "@babel/preset-env": "7.2.0",
  "babel-eslint": "5.0.0",
  "babel-jest": "8.0.0",
  "babel-loader": "5.0.0",
  "eslint": "5.10.0",
  "eslint-plugin-jest": "8.0.0",
  "jest": "5.0.0",
  "webpack": "4.28.3",
  "webpack-cli": "3.2.1",
}
`;

exports[`CTF combination babel,webpack,eslint,prettier 1`] = `
Array [
  "babel",
  "webpack",
  "eslint",
  "prettier",
]
`;

exports[`CTF combination babel,webpack,eslint,prettier 2`] = `
Array [
  Object {
    "extends": Array [
      "@babel/preset-env",
    ],
  },
  Object {
    "entry": "/",
    "mode": "development",
    "module": Object {
      "rules": Array [
        Object {
          "exclude": /node_modules/,
          "test": /\\\\\\.jsx\\?\\$/,
          "use": Object {
            "loader": "babel-loader",
            "options": Object {
              "cacheDirectory": true,
            },
          },
        },
      ],
    },
    "output": Object {
      "filename": "main.js",
      "path": "/",
      "publicPath": "./targets/",
    },
    "plugins": Array [],
    "resolve": Object {
      "extensions": Array [
        ".js",
        ".json",
      ],
    },
  },
  Object {
    "extends": Array [
      "prettier",
    ],
    "parser": "babel-eslint",
    "plugins": Array [
      "prettier",
    ],
    "settings": Object {
      "import/resolver": Object {
        "webpack": Object {
          "config": "configs/webpack.config.js",
        },
      },
    },
  },
  Object {},
]
`;

exports[`CTF combination babel,webpack,eslint,prettier 3`] = `
Object {
  "eslint-config-prettier": "3.3.0",
  "eslint-plugin-prettier": "3.0.1",
}
`;

exports[`CTF combination babel,webpack,eslint,prettier 4`] = `
Object {
  "@babel/cli": "7.2.0",
  "@babel/core": "7.2.0",
  "@babel/preset-env": "7.2.0",
  "babel-eslint": "5.0.0",
  "babel-loader": "5.0.0",
  "eslint": "5.10.0",
  "prettier": "1.15.3",
  "webpack": "4.28.3",
  "webpack-cli": "3.2.1",
}
`;

exports[`CTF combination babel,webpack,eslint,prettier,jest 1`] = `
Array [
  "babel",
  "webpack",
  "eslint",
  "prettier",
  "jest",
]
`;

exports[`CTF combination babel,webpack,eslint,prettier,jest 2`] = `
Array [
  Object {
    "extends": Array [
      "@babel/preset-env",
    ],
  },
  Object {
    "entry": "/",
    "mode": "development",
    "module": Object {
      "rules": Array [
        Object {
          "exclude": /node_modules/,
          "test": /\\\\\\.jsx\\?\\$/,
          "use": Object {
            "loader": "babel-loader",
            "options": Object {
              "cacheDirectory": true,
            },
          },
        },
      ],
    },
    "output": Object {
      "filename": "main.js",
      "path": "/",
      "publicPath": "./targets/",
    },
    "plugins": Array [],
    "resolve": Object {
      "extensions": Array [
        ".js",
        ".json",
      ],
    },
  },
  Object {
    "extends": Array [
      "prettier",
    ],
    "parser": "babel-eslint",
    "plugins": Array [
      "jest",
    ],
    "settings": Object {
      "import/resolver": Object {
        "webpack": Object {
          "config": "configs/webpack.config.js",
        },
      },
    },
  },
  Object {},
  Object {
    "moduleNameMapper": Object {
      "\\\\.(css|less|sass|scss)$": "identity-obj-proxy",
      "\\\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/mocks/fileMock.js",
    },
  },
]
`;

exports[`CTF combination babel,webpack,eslint,prettier,jest 3`] = `
Object {
  "eslint-config-prettier": "3.3.0",
  "eslint-plugin-prettier": "3.0.1",
}
`;

exports[`CTF combination babel,webpack,eslint,prettier,jest 4`] = `
Object {
  "@babel/cli": "7.2.0",
  "@babel/core": "7.2.0",
  "@babel/preset-env": "7.2.0",
  "babel-eslint": "5.0.0",
  "babel-jest": "8.0.0",
  "babel-loader": "5.0.0",
  "eslint": "5.10.0",
  "eslint-plugin-jest": "8.0.0",
  "jest": "5.0.0",
  "prettier": "1.15.3",
  "webpack": "4.28.3",
  "webpack-cli": "3.2.1",
}
`;

exports[`CTF combination babel,webpack,jest 1`] = `
Array [
  "babel",
  "webpack",
  "jest",
]
`;

exports[`CTF combination babel,webpack,jest 2`] = `
Array [
  Object {
    "extends": Array [
      "@babel/preset-env",
    ],
  },
  Object {
    "entry": "/",
    "mode": "development",
    "module": Object {
      "rules": Array [
        Object {
          "exclude": /node_modules/,
          "test": /\\\\\\.jsx\\?\\$/,
          "use": Object {
            "loader": "babel-loader",
            "options": Object {
              "cacheDirectory": true,
            },
          },
        },
      ],
    },
    "output": Object {
      "filename": "main.js",
      "path": "/",
      "publicPath": "./targets/",
    },
    "plugins": Array [],
    "resolve": Object {
      "extensions": Array [
        ".js",
        ".json",
      ],
    },
  },
  Object {
    "moduleNameMapper": Object {
      "\\\\.(css|less|sass|scss)$": "identity-obj-proxy",
      "\\\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/mocks/fileMock.js",
    },
  },
]
`;

exports[`CTF combination babel,webpack,jest 3`] = `Object {}`;

exports[`CTF combination babel,webpack,jest 4`] = `
Object {
  "@babel/cli": "7.2.0",
  "@babel/core": "7.2.0",
  "@babel/preset-env": "7.2.0",
  "babel-jest": "8.0.0",
  "babel-loader": "5.0.0",
  "jest": "5.0.0",
  "webpack": "4.28.3",
  "webpack-cli": "3.2.1",
}
`;

exports[`CTF combination babel,webpack,prettier 1`] = `
Array [
  "babel",
  "webpack",
  "prettier",
]
`;

exports[`CTF combination babel,webpack,prettier 2`] = `
Array [
  Object {
    "extends": Array [
      "@babel/preset-env",
    ],
  },
  Object {
    "entry": "/",
    "mode": "development",
    "module": Object {
      "rules": Array [
        Object {
          "exclude": /node_modules/,
          "test": /\\\\\\.jsx\\?\\$/,
          "use": Object {
            "loader": "babel-loader",
            "options": Object {
              "cacheDirectory": true,
            },
          },
        },
      ],
    },
    "output": Object {
      "filename": "main.js",
      "path": "/",
      "publicPath": "./targets/",
    },
    "plugins": Array [],
    "resolve": Object {
      "extensions": Array [
        ".js",
        ".json",
      ],
    },
  },
  Object {},
]
`;

exports[`CTF combination babel,webpack,prettier 3`] = `Object {}`;

exports[`CTF combination babel,webpack,prettier 4`] = `
Object {
  "@babel/cli": "7.2.0",
  "@babel/core": "7.2.0",
  "@babel/preset-env": "7.2.0",
  "babel-loader": "5.0.0",
  "prettier": "1.15.3",
  "webpack": "4.28.3",
  "webpack-cli": "3.2.1",
}
`;

exports[`CTF combination babel,webpack,prettier,jest 1`] = `
Array [
  "babel",
  "webpack",
  "prettier",
  "jest",
]
`;

exports[`CTF combination babel,webpack,prettier,jest 2`] = `
Array [
  Object {
    "extends": Array [
      "@babel/preset-env",
    ],
  },
  Object {
    "entry": "/",
    "mode": "development",
    "module": Object {
      "rules": Array [
        Object {
          "exclude": /node_modules/,
          "test": /\\\\\\.jsx\\?\\$/,
          "use": Object {
            "loader": "babel-loader",
            "options": Object {
              "cacheDirectory": true,
            },
          },
        },
      ],
    },
    "output": Object {
      "filename": "main.js",
      "path": "/",
      "publicPath": "./targets/",
    },
    "plugins": Array [],
    "resolve": Object {
      "extensions": Array [
        ".js",
        ".json",
      ],
    },
  },
  Object {},
  Object {
    "moduleNameMapper": Object {
      "\\\\.(css|less|sass|scss)$": "identity-obj-proxy",
      "\\\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/mocks/fileMock.js",
    },
  },
]
`;

exports[`CTF combination babel,webpack,prettier,jest 3`] = `Object {}`;

exports[`CTF combination babel,webpack,prettier,jest 4`] = `
Object {
  "@babel/cli": "7.2.0",
  "@babel/core": "7.2.0",
  "@babel/preset-env": "7.2.0",
  "babel-jest": "8.0.0",
  "babel-loader": "5.0.0",
  "jest": "5.0.0",
  "prettier": "1.15.3",
  "webpack": "4.28.3",
  "webpack-cli": "3.2.1",
}
`;

exports[`CTF combination eslint 1`] = `
Array [
  "eslint",
]
`;

exports[`CTF combination eslint 2`] = `
Array [
  Object {
    "extends": Array [
      "bliss",
    ],
  },
]
`;

exports[`CTF combination eslint 3`] = `Object {}`;

exports[`CTF combination eslint 4`] = `
Object {
  "eslint": "5.10.0",
}
`;

exports[`CTF combination eslint,jest 1`] = `
Array [
  "eslint",
  "jest",
]
`;

exports[`CTF combination eslint,jest 2`] = `
Array [
  Object {
    "extends": Array [
      "bliss",
    ],
    "plugins": Array [
      "jest",
    ],
  },
  Object {},
]
`;

exports[`CTF combination eslint,jest 3`] = `Object {}`;

exports[`CTF combination eslint,jest 4`] = `
Object {
  "eslint": "5.10.0",
  "eslint-plugin-jest": "8.0.0",
  "jest": "5.0.0",
}
`;

exports[`CTF combination eslint,prettier 1`] = `
Array [
  "eslint",
  "prettier",
]
`;

exports[`CTF combination eslint,prettier 2`] = `
Array [
  Object {
    "extends": Array [
      "prettier",
    ],
    "plugins": Array [
      "prettier",
    ],
  },
  Object {},
]
`;

exports[`CTF combination eslint,prettier 3`] = `
Object {
  "eslint-config-prettier": "3.3.0",
  "eslint-plugin-prettier": "3.0.1",
}
`;

exports[`CTF combination eslint,prettier 4`] = `
Object {
  "eslint": "5.10.0",
  "prettier": "1.15.3",
}
`;

exports[`CTF combination eslint,prettier,jest 1`] = `
Array [
  "eslint",
  "prettier",
  "jest",
]
`;

exports[`CTF combination eslint,prettier,jest 2`] = `
Array [
  Object {
    "extends": Array [
      "prettier",
    ],
    "plugins": Array [
      "jest",
    ],
  },
  Object {},
  Object {},
]
`;

exports[`CTF combination eslint,prettier,jest 3`] = `
Object {
  "eslint-config-prettier": "3.3.0",
  "eslint-plugin-prettier": "3.0.1",
}
`;

exports[`CTF combination eslint,prettier,jest 4`] = `
Object {
  "eslint": "5.10.0",
  "eslint-plugin-jest": "8.0.0",
  "jest": "5.0.0",
  "prettier": "1.15.3",
}
`;

exports[`CTF combination jest 1`] = `
Array [
  "jest",
]
`;

exports[`CTF combination jest 2`] = `
Array [
  Object {},
]
`;

exports[`CTF combination jest 3`] = `Object {}`;

exports[`CTF combination jest 4`] = `
Object {
  "jest": "5.0.0",
}
`;

exports[`CTF combination prettier 1`] = `
Array [
  "prettier",
]
`;

exports[`CTF combination prettier 2`] = `
Array [
  Object {},
]
`;

exports[`CTF combination prettier 3`] = `Object {}`;

exports[`CTF combination prettier 4`] = `
Object {
  "prettier": "1.15.3",
}
`;

exports[`CTF combination prettier,jest 1`] = `
Array [
  "prettier",
  "jest",
]
`;

exports[`CTF combination prettier,jest 2`] = `
Array [
  Object {},
  Object {},
]
`;

exports[`CTF combination prettier,jest 3`] = `Object {}`;

exports[`CTF combination prettier,jest 4`] = `
Object {
  "jest": "5.0.0",
  "prettier": "1.15.3",
}
`;

exports[`CTF combination webpack 1`] = `
Array [
  "webpack",
]
`;

exports[`CTF combination webpack 2`] = `
Array [
  Object {
    "entry": "/",
    "mode": "development",
    "output": Object {
      "filename": "main.js",
      "path": "/",
      "publicPath": "./targets/",
    },
    "plugins": Array [],
    "resolve": Object {
      "extensions": Array [
        ".js",
        ".json",
      ],
    },
  },
]
`;

exports[`CTF combination webpack 3`] = `Object {}`;

exports[`CTF combination webpack 4`] = `
Object {
  "webpack": "4.28.3",
  "webpack-cli": "3.2.1",
}
`;

exports[`CTF combination webpack,eslint 1`] = `
Array [
  "webpack",
  "eslint",
]
`;

exports[`CTF combination webpack,eslint 2`] = `
Array [
  Object {
    "entry": "/",
    "mode": "development",
    "output": Object {
      "filename": "main.js",
      "path": "/",
      "publicPath": "./targets/",
    },
    "plugins": Array [],
    "resolve": Object {
      "extensions": Array [
        ".js",
        ".json",
      ],
    },
  },
  Object {
    "extends": Array [
      "bliss",
    ],
    "settings": Object {
      "import/resolver": Object {
        "webpack": Object {
          "config": "configs/webpack.config.js",
        },
      },
    },
  },
]
`;

exports[`CTF combination webpack,eslint 3`] = `Object {}`;

exports[`CTF combination webpack,eslint 4`] = `
Object {
  "eslint": "5.10.0",
  "webpack": "4.28.3",
  "webpack-cli": "3.2.1",
}
`;

exports[`CTF combination webpack,eslint,jest 1`] = `
Array [
  "webpack",
  "eslint",
  "jest",
]
`;

exports[`CTF combination webpack,eslint,jest 2`] = `
Array [
  Object {
    "entry": "/",
    "mode": "development",
    "output": Object {
      "filename": "main.js",
      "path": "/",
      "publicPath": "./targets/",
    },
    "plugins": Array [],
    "resolve": Object {
      "extensions": Array [
        ".js",
        ".json",
      ],
    },
  },
  Object {
    "extends": Array [
      "bliss",
    ],
    "plugins": Array [
      "jest",
    ],
    "settings": Object {
      "import/resolver": Object {
        "webpack": Object {
          "config": "configs/webpack.config.js",
        },
      },
    },
  },
  Object {
    "moduleNameMapper": Object {
      "\\\\.(css|less|sass|scss)$": "identity-obj-proxy",
      "\\\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/mocks/fileMock.js",
    },
  },
]
`;

exports[`CTF combination webpack,eslint,jest 3`] = `Object {}`;

exports[`CTF combination webpack,eslint,jest 4`] = `
Object {
  "eslint": "5.10.0",
  "eslint-plugin-jest": "8.0.0",
  "jest": "5.0.0",
  "webpack": "4.28.3",
  "webpack-cli": "3.2.1",
}
`;

exports[`CTF combination webpack,eslint,prettier 1`] = `
Array [
  "webpack",
  "eslint",
  "prettier",
]
`;

exports[`CTF combination webpack,eslint,prettier 2`] = `
Array [
  Object {
    "entry": "/",
    "mode": "development",
    "output": Object {
      "filename": "main.js",
      "path": "/",
      "publicPath": "./targets/",
    },
    "plugins": Array [],
    "resolve": Object {
      "extensions": Array [
        ".js",
        ".json",
      ],
    },
  },
  Object {
    "extends": Array [
      "prettier",
    ],
    "plugins": Array [
      "prettier",
    ],
    "settings": Object {
      "import/resolver": Object {
        "webpack": Object {
          "config": "configs/webpack.config.js",
        },
      },
    },
  },
  Object {},
]
`;

exports[`CTF combination webpack,eslint,prettier 3`] = `
Object {
  "eslint-config-prettier": "3.3.0",
  "eslint-plugin-prettier": "3.0.1",
}
`;

exports[`CTF combination webpack,eslint,prettier 4`] = `
Object {
  "eslint": "5.10.0",
  "prettier": "1.15.3",
  "webpack": "4.28.3",
  "webpack-cli": "3.2.1",
}
`;

exports[`CTF combination webpack,eslint,prettier,jest 1`] = `
Array [
  "webpack",
  "eslint",
  "prettier",
  "jest",
]
`;

exports[`CTF combination webpack,eslint,prettier,jest 2`] = `
Array [
  Object {
    "entry": "/",
    "mode": "development",
    "output": Object {
      "filename": "main.js",
      "path": "/",
      "publicPath": "./targets/",
    },
    "plugins": Array [],
    "resolve": Object {
      "extensions": Array [
        ".js",
        ".json",
      ],
    },
  },
  Object {
    "extends": Array [
      "prettier",
    ],
    "plugins": Array [
      "jest",
    ],
    "settings": Object {
      "import/resolver": Object {
        "webpack": Object {
          "config": "configs/webpack.config.js",
        },
      },
    },
  },
  Object {},
  Object {
    "moduleNameMapper": Object {
      "\\\\.(css|less|sass|scss)$": "identity-obj-proxy",
      "\\\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/mocks/fileMock.js",
    },
  },
]
`;

exports[`CTF combination webpack,eslint,prettier,jest 3`] = `
Object {
  "eslint-config-prettier": "3.3.0",
  "eslint-plugin-prettier": "3.0.1",
}
`;

exports[`CTF combination webpack,eslint,prettier,jest 4`] = `
Object {
  "eslint": "5.10.0",
  "eslint-plugin-jest": "8.0.0",
  "jest": "5.0.0",
  "prettier": "1.15.3",
  "webpack": "4.28.3",
  "webpack-cli": "3.2.1",
}
`;

exports[`CTF combination webpack,jest 1`] = `
Array [
  "webpack",
  "jest",
]
`;

exports[`CTF combination webpack,jest 2`] = `
Array [
  Object {
    "entry": "/",
    "mode": "development",
    "output": Object {
      "filename": "main.js",
      "path": "/",
      "publicPath": "./targets/",
    },
    "plugins": Array [],
    "resolve": Object {
      "extensions": Array [
        ".js",
        ".json",
      ],
    },
  },
  Object {
    "moduleNameMapper": Object {
      "\\\\.(css|less|sass|scss)$": "identity-obj-proxy",
      "\\\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/mocks/fileMock.js",
    },
  },
]
`;

exports[`CTF combination webpack,jest 3`] = `Object {}`;

exports[`CTF combination webpack,jest 4`] = `
Object {
  "jest": "5.0.0",
  "webpack": "4.28.3",
  "webpack-cli": "3.2.1",
}
`;

exports[`CTF combination webpack,prettier 1`] = `
Array [
  "webpack",
  "prettier",
]
`;

exports[`CTF combination webpack,prettier 2`] = `
Array [
  Object {
    "entry": "/",
    "mode": "development",
    "output": Object {
      "filename": "main.js",
      "path": "/",
      "publicPath": "./targets/",
    },
    "plugins": Array [],
    "resolve": Object {
      "extensions": Array [
        ".js",
        ".json",
      ],
    },
  },
  Object {},
]
`;

exports[`CTF combination webpack,prettier 3`] = `Object {}`;

exports[`CTF combination webpack,prettier 4`] = `
Object {
  "prettier": "1.15.3",
  "webpack": "4.28.3",
  "webpack-cli": "3.2.1",
}
`;

exports[`CTF combination webpack,prettier,jest 1`] = `
Array [
  "webpack",
  "prettier",
  "jest",
]
`;

exports[`CTF combination webpack,prettier,jest 2`] = `
Array [
  Object {
    "entry": "/",
    "mode": "development",
    "output": Object {
      "filename": "main.js",
      "path": "/",
      "publicPath": "./targets/",
    },
    "plugins": Array [],
    "resolve": Object {
      "extensions": Array [
        ".js",
        ".json",
      ],
    },
  },
  Object {},
  Object {
    "moduleNameMapper": Object {
      "\\\\.(css|less|sass|scss)$": "identity-obj-proxy",
      "\\\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/mocks/fileMock.js",
    },
  },
]
`;

exports[`CTF combination webpack,prettier,jest 3`] = `Object {}`;

exports[`CTF combination webpack,prettier,jest 4`] = `
Object {
  "jest": "5.0.0",
  "prettier": "1.15.3",
  "webpack": "4.28.3",
  "webpack-cli": "3.2.1",
}
`;

exports[`CTF executors should generate functions for scripts 1`] = `
Object {
  "build": [Function],
}
`;

exports[`CTF should add devDepencencies 1`] = `
Object {
  "foobar": "0.0.0",
  "webpack": "4.28.3",
  "webpack-cli": "3.2.1",
}
`;
